### ISSUE TITLE
Support webhook_url_file for discord config

### ISSUE THREAD

**Issue by josedev-union** [prometheus/alertmanager#3482]
**What did you do?**
I use a main Discord for receiver.
**What did you expect to see?**
I want to use pre-existing K8s secret to configure Discord receiver's webhook url instead of setting that as helm value. Like `slack_api_url_file`, mount the existing k8s secret into alertmanager and use that file as `webhook_url`.

**Comment by Roronoa-Zoroo** [comment]
the documentation mentions the use of webhook_url_file for discord_config
i was pulling my hair as to why it was not working

**Comment by arichtman** [comment]
The configuration documentation does mention this option, but when I try to provide just `webhook_url_file` it complains that no `webhook_url` is provided, and if I provide both, it exclaims that only one is permissible.

**Comment by tuana9a** [comment]
@arichtman same here, just found out that this has been fixed in #3728 but it hasn't been released yet.

**Comment by arichtman** [comment]
v0.28.0 seems to have the fix.

**Comment by iliesbenyahia** [comment]
Hello, is there any progress about this issue ? i've been strugling all day to make it work, only to realize that it is in fact an issue

**Comment by arichtman** [comment]
@iliesbenyahia 

There are several updates on this just above your comment from November.

If you are experiencing the bug please confirm which version you are using.

**Comment by denis-volin** [comment]
@arichtman 
Still a thing in v0.30.0. (I'm using kube-prometheus-stack Helm chart version 80.13.3)

```
provision alertmanager configuration: failed to initialize from secret: yaml: unmarshal errors:
  line 30: field webhook_url_file not found in type alertmanager.discordConfig
```

**Comment by TheMeier** [comment]
> [@arichtman](https://github.com/arichtman) Still a thing in v0.30.0. (I'm using kube-prometheus-stack Helm chart version 80.13.3)
> 
> ```
> provision alertmanager configuration: failed to initialize from secret: yaml: unmarshal errors:
>   line 30: field webhook_url_file not found in type alertmanager.discordConfig
> ```

Isn't that an issue with the operator?
https://github.com/prometheus-operator/prometheus-operator/blob/main/pkg/alertmanager/types.go#L383

**Comment by TheMeier** [comment]
IMHO this is done

**Comment by denis-volin** [comment]
> > [@arichtman](https://github.com/arichtman) Still a thing in v0.30.0. (I'm using kube-prometheus-stack Helm chart version 80.13.3)
> > ```
> > provision alertmanager configuration: failed to initialize from secret: yaml: unmarshal errors:
> >   line 30: field webhook_url_file not found in type alertmanager.discordConfig
> > ```
> 
> Isn't that an issue with the operator? https://github.com/prometheus-operator/prometheus-operator/blob/main/pkg/alertmanager/types.go#L383

Yeah, I even found a few PRs on that topic. Seems like it's so unpopular, that even after 2 years this is still hasn't got into the main branch...

**Comment by TheMeier** [comment]
Still I think in alertmanager this is  completed. Please re-open this issue if I am wrong.
